app/services/weather_request_service.rb

```diff
@@ -0,0 +1,61 @@
+require 'net/http'
+require 'json'
+
+class WeatherRequestService
+
+  def initialize(location)
+    @location        = location
+    @weather_request = setup_request(@location)
+  end
+
+  def run
+    call_weather_api
+    hash_data = try_json_to_hash                       if @weather_request.status == WeatherRequest::PROCESSING
+    validate_data(hash_data)                           if @weather_request.status == WeatherRequest::PROCESSING
+    @weather_request.status = WeatherRequest::SUCCESS  if @weather_request.status == WeatherRequest::PROCESSING
+    report_error                                       if @weather_request.status == WeatherRequest::ERROR
+    @weather_request.save
+    @weather_request
+  end
+
+  private
+
+    def setup_request(location)
+      request_url = Rails.configuration.yahoo_weather_url.sub('LOCATION_ID', location.yahoo_id)
+      WeatherRequest.create(location: location, request_url: request_url, status: WeatherRequest::PROCESSING)
```

Comment:
Because `status` is a heavily used attribute, I would make it more visible: e.g.
```ruby
      WeatherRequest.create(
        status:         WeatherRequest::PROCESSING,
        location:       location,
        request_url:    request_url
      )
```
What do you think?

Reply:
I agree.  Thanks.